services/vm-watcher/vm_watcher_service.go

```diff
@@ -144,6 +155,8 @@ func (s Service) vmFolderMoRefWithIDs(
 			}
 			moids[v] = append(moids[v], fmt.Sprintf("%s/%s", z.Namespace, z.Name))
 
+			// API server prevents adding finalizers to resources that
+			// are in Deleting state, so skip such zones.
```

Comment:
This comment seems a little out of place - that check & skip is done up above. Here, we must make sure that the finalizer is there for any Zones that are apart of the `moids` map